tests/utils/state_locks_filter.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+import argparse
+import re
+import sys
+from typing import IO
+
+class LockOpTrace:
+    """
+    LockOpTrace: Represents a Lock Operation trace in the log file.
+    It holds the lines for the trace and allows matching against a part of the trace.
+    """
+
+    def __init__(self, lines: list[str]):
+        self.lines = lines
+        self.hash = hash(str(self))
+
+    def get_trace_lines(self) -> list[str]:
+        return self.lines
+
+    def match(self, part: str) -> bool:
+        for line in self.lines:
+            if part in line:
+                return True
+
+        return False
+
+    def __str__(self) -> str:
+        return "".join(self.lines)
+
+    def __hash__(self) -> int:
+        return self.hash
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, LockOpTrace):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+
+        return self.hash == other.hash
+
+
+
+class LockOp:
+    """
+    LockOp: Represents a lock operation in the log file, including its header and times (held and wait times).
+    It is tied to the LockOpTrace class.
+    """
+
+    def __init__(self, lines: list[str]):
+        self.trace = LockOpTrace(lines[1:])
+        self.header = lines[0]
+        self.held_time = 0
+        self.wait_time = 0
+        self._calc_held_ms(self.header)
+        self._calc_wait_ms(self.header)
+
+    def _calc_held_ms(self, line: str) -> int:
```

Comment:
Since the granularity is in milliseconds can we use datetime.timedelta for this to make it clear that this is a time-difference value and what the value 10 means?